tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -79,6 +79,67 @@ def test_slurm_gpu(region, pcluster_config_reader, clusters_factory):
     assert_no_errors_in_logs(remote_command_executor, ["/var/log/sqswatcher", "/var/log/jobwatcher"])
 
 
+@pytest.mark.regions(["us-east-2"])
+@pytest.mark.instances(["c5.xlarge"])
+@pytest.mark.schedulers(["slurm"])
+@pytest.mark.skip_oss(["centos6"])
+@pytest.mark.usefixtures("os", "instance", "scheduler")
+@pytest.mark.slurm_mpi
+def test_slurm_mpi(region, pcluster_config_reader, clusters_factory, test_datadir):
+    """
+    Test Slurm with mpirun.
+
+    mpirun job is known to leave stray processes after job termination if slurm process tracking is not setup correctly,
+    i.e. using ProctrackType=proctrack/pgid
+    Test mpirun script to make sure no stray processes after the job is terminated due to timeout
+    Test should run on all OSs except for centos6, where mpirun is not available
+    """
+    cluster_config = pcluster_config_reader()
+    cluster = clusters_factory(cluster_config)
+    remote_command_executor = RemoteCommandExecutor(cluster)
+
+    _test_mpi_job_termination(remote_command_executor, test_datadir)
+
+    assert_no_errors_in_logs(remote_command_executor, ["/var/log/sqswatcher", "/var/log/jobwatcher"])
+
+
+def _test_mpi_job_termination(remote_command_executor, test_datadir):
+    """Test canceling mpirun job will not leave stray processes."""
+    logging.info("Testing no stray process left behind after mpirun job is terminated")
+    slurm_commands = SlurmCommands(remote_command_executor)
+
+    # Submit mpi_job, leaving 1 vcpu on each node idle so that the process check job can run while mpi_job is running
+    # Note: using 5 compute nodes for mpi_job so that the job takes more time to complete than the jobs's time limit
```

Comment:
can we use fewer compute nodes? (1 would be ideal) and maybe issue an scancel if we can't rely on the timeout?